examples/asr_training/rnnt_lightning.py

```diff
@@ -0,0 +1,535 @@
+    hypos_str = [sp_model.DecodePieces(s.split()) for s in hypos_str]
+    hypos_ali = [h.alignment[1:] for h in hypos]
+    hypos_ids = [h.tokens[1:] for h in hypos]
+    hypos_score = [[math.exp(h.score)] for h in hypos]
+
+    nbest_batch = list(zip(hypos_str, hypos_score, hypos_ali, hypos_ids))
+
+    return nbest_batch
+
+
+def run_eval(args):
+    model = (
+        RNNTModule.load_from_checkpoint(
+            args.checkpoint_path,
+            librispeech_path=args.librispeech_path,
+            sp_model_path=args.sp_model_path,
+            tgt_dict_path=args.tgt_dict_path,
+            global_stats_path=args.global_stats_path,
+        )
+        .eval()
+        .to(device="cuda")
+    )
+    total_edit_distance = 0
+    total_length = 0
+    dataloader = model.test_dataloader()
+    with torch.no_grad():
+        for idx, (batch, sample) in enumerate(dataloader):
+            actual = sample[0][2]
+            predicted = model(batch)
+            total_edit_distance += compute_word_level_distance(actual, predicted)
+            total_length += len(actual.split())
+            if idx % 100 == 0:
+                logger.info(
+                    f"Processed elem {idx}; WER: {total_edit_distance / total_length}"
+                )
+    logger.info(f"Final WER: {total_edit_distance / total_length}")
+
+
+def run_train(args):
+    checkpoint_dir = args.exp_dir / "checkpoints"
+    checkpoint = ModelCheckpoint(
+        checkpoint_dir,
+        monitor="Losses/val_loss",
+        mode="min",
+        save_top_k=10,
+        save_weights_only=True,
+        verbose=True,
+    )
+    train_checkpoint = ModelCheckpoint(
+        checkpoint_dir,
+        monitor="Losses/train_loss",
+        mode="min",
+        save_top_k=10,
+        save_weights_only=True,
+        verbose=True,
+    )
+    callbacks = [
+        checkpoint,
+        train_checkpoint,
+    ]
+    trainer = Trainer(
+        default_root_dir=args.exp_dir,
+        max_epochs=args.epochs,
+        num_nodes=args.num_nodes,
+        gpus=args.gpus,
+        accelerator="gpu",
+        strategy="ddp",
+        gradient_clip_val=10.0,
+        callbacks=callbacks,
+    )
+
+    model = RNNTModule(
+        librispeech_path=args.librispeech_path,
+        sp_model_path=args.sp_model_path,
+        tgt_dict_path=args.tgt_dict_path,
+        global_stats_path=args.global_stats_path,
+    )
+    trainer.fit(model)
+
+
+def cli_main():
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--eval", action="store_true", default=False, help="Run in eval mode.",
+    )
+    parser.add_argument(
+        "--exp_dir",
+        default=pathlib.Path("./exp"),
+        type=pathlib.Path,
+        help="Directory to save checkpoints and logs to. (Default: './exp')",
+    )
+    parser.add_argument(
+        "--checkpoint_path",
+        type=pathlib.Path,
+        help="Path to checkpoint to use for evaluation.",
+    )
+    parser.add_argument(
+        "--global_stats_path",
```

Comment:
How to get the means and stds for the training features?

Reply:
i can include the json file i'm currently using. later on, perhaps i can introduce a utility that shows how that file is generated